AppDashboard/dashboard.py

```diff
@@ -1128,12 +1204,57 @@ def get(self):
       'result' : self.helper.run_groomer()
     }))
 
+class AjaxRenderPanel(AppDashboard):
+  """ Class that adds panels to the dashboard. """
+
+  def post(self):
+    """ Calls render_template to return the correct panel """
+    key_val = self.request.get('key_val')
+    self.response.out.write(self.render_template(
+      template_file='layouts/panel.html',
+      values={'page_info':self.dstore.build_dict()[key_val],
+              'id':key_val}))
+
+class AjaxSaveLayoutSettings(AppDashboard):
+  """ Class that stores dashboard layout settings in the Datastore. """
+
+  def post(self):
+    """ sets the dashboard layout settings """
+
+    nav = self.request.get("nav")
+    panel = self.request.get("panel")
+    json_string = '{"nav":'+nav+',"panel":'+panel+"}"
```

Comment:
In this case, I think it's better to fail fast and let the client know that their request was invalid. Otherwise, if for some reason there's a JavaScript bug somewhere down the line, it could look like the post is succeeding but having no effect. It would be clearer if the response from the post returned a 400 or something.